### ISSUE TITLE
[ios] Animation doesn't play if system dark/light mode is different to that of the app

### ISSUE THREAD

**Issue by cjpete** [lottie-react-native/lottie-react-native#954]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `lottie-react-native@5.1.4` for the project I'm working on.

### Description of issue 

When the system color mode trait is different to that of the color mode of the app, the `traitCollectionDidChange` callback is triggered on mount. The call to `applyProperties` then causes the animation to not play (even though autoPlay is set).

You can set the color mode of the app by using `window.overrideUserInterfaceStyle = .dark`.

### Here is the diff that solved my problem:

```diff
diff --git a/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift b/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
index cd8ed10..338b90b 100644
--- a/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
+++ b/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
@@ -17,7 +17,9 @@ class ContainerView: RCTView {
         super.traitCollectionDidChange(previousTraitCollection)
         if #available(iOS 13.0, tvOS 13.0, *) {
             if (self.traitCollection.hasDifferentColorAppearance(comparedTo: previousTraitCollection)) {
-                applyProperties()
+                if (colorFilters.count > 0) {
+                    applyProperties()
+                }
                 print("dark mode changed")
             }
         }
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by wangliang1124** [comment]
can reproduce with 5.1.4
here is my fix:

`
diff --git a/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift b/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
index cd8ed10..8a829b4 100644
--- a/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
+++ b/node_modules/lottie-react-native/src/ios/LottieReactNative/ContainerView.swift
@@ -17,7 +17,7 @@ class ContainerView: RCTView {
         super.traitCollectionDidChange(previousTraitCollection)
         if #available(iOS 13.0, tvOS 13.0, *) {
             if (self.traitCollection.hasDifferentColorAppearance(comparedTo: previousTraitCollection)) {
-                applyProperties()
+                applyColorProperties()
                 print("dark mode changed")
             }
         }
@@ -213,4 +213,17 @@ class ContainerView: RCTView {
            resume()
         }
     }
+
+     func applyColorProperties() {
+        guard let animationView = animationView else { return }
+
+        if (colorFilters.count > 0) {
+            for filter in colorFilters {
+                let keypath: String = "\(filter.value(forKey: "keypath") as! String).**.Color"
+                let fillKeypath = AnimationKeypath(keypath: keypath)
+                let colorFilterValueProvider = ColorValueProvider((filter.value(forKey: "color") as! PlatformColor).lottieColorValue)
+                animationView.setValueProvider(colorFilterValueProvider, keypath: fillKeypath)
+            }
+        }
+    }
 }

`

**Comment by syou98** [comment]
yes I tried this morning, and I still have the issue

**Comment by TheRogue76** [comment]
Heya!

I took a look into the repo and it looks like a change similar to the one recommended here was applied to the repo a while ago:

```swift
#if !(os(OSX))
    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
        super.traitCollectionDidChange(previousTraitCollection)
        if #available(iOS 13.0, tvOS 13.0, *) {
            if self.traitCollection.hasDifferentColorAppearance(comparedTo: previousTraitCollection) {
                if !colorFilters.isEmpty {
                    applyColorProperties()
                }
            }
        }
    }
#endif
```

Can someone check 6.5.1 to see if the issue has been resolved? If it has, great, I'll close it. If not, I can work on it and get it done.

**Comment by TheRogue76** [comment]
Closing due to inactivity
